patches/minecraft/net/minecraft/block/BlockCactus.java.patch

```diff
@@ -9,7 +9,41 @@
  {
      public static final PropertyInteger field_176587_a = PropertyInteger.func_177719_a("age", 0, 15);
      protected static final AxisAlignedBB field_185593_b = new AxisAlignedBB(0.0625D, 0.0D, 0.0625D, 0.9375D, 0.9375D, 0.9375D);
-@@ -110,8 +110,8 @@
+@@ -48,18 +48,22 @@
+             if (i < 3)
+             {
+                 int j = ((Integer)p_180650_3_.func_177229_b(field_176587_a)).intValue();
+-
+-                if (j == 15)
+-                {
+-                    p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
+-                    IBlockState iblockstate = p_180650_3_.func_177226_a(field_176587_a, Integer.valueOf(0));
+-                    p_180650_1_.func_180501_a(p_180650_2_, iblockstate, 4);
+-                    iblockstate.func_189546_a(p_180650_1_, blockpos, this);
++                net.minecraftforge.event.world.BlockEvent ev = new net.minecraftforge.event.world.BlockEvent.CropGrowEvent.Pre(p_180650_1_,p_180650_2_,p_180650_3_);
++                net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(ev);
++                if(ev.getResult() != net.minecraftforge.fml.common.eventhandler.Event.Result.DENY) {
++                    if (j == 15)
++                    {
++                        p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
```

Comment:
This seems to be the same paragraph as before, just indented.
Just leave it unindented and change only necessary lines.